phlib/mapldr.c

```diff
@@ -682,18 +682,18 @@ NTSTATUS PhLoadResource(
     )
 {
     NTSTATUS status;
-    LDR_RESOURCE_INFO resourceInfo;
+    LDR_EXTERNAL_DLL_PATH resourceIdPath[3];
     PIMAGE_RESOURCE_DATA_ENTRY resourceData = NULL;
     PVOID resourceBuffer = NULL;
     ULONG resourceLength;
 
-    resourceInfo.Type = (ULONG_PTR)Type;
-    resourceInfo.Name = (ULONG_PTR)Name;
-    resourceInfo.Language = MAKELANGID(LANG_NEUTRAL, SUBLANG_NEUTRAL);
+    resourceIdPath[LDR_RESOURCE_PATH_TYPE] = (ULONG_PTR)Type;
+    resourceIdPath[LDR_RESOURCE_PATH_NAME] = (ULONG_PTR)Name;
+    resourceIdPath[LDR_RESOURCE_PATH_LANGUAGE] = MAKELANGID(LANG_NEUTRAL, SUBLANG_NEUTRAL);
 
     __try
     {
-        status = LdrFindResource_U(DllBase, &resourceInfo, RESOURCE_DATA_LEVEL, &resourceData);
+        status = LdrFindResource_U(DllBase, resourceIdPath, RTL_NUMBER_OF(resourceIdPath), &resourceData);
```

Comment:
I don't think these can be in an arbitrary order? There is an explicit leveling to locating a resource. So having the `LDR_RESOURCE_INFO` be explicit seems most convenient to me. I'd also like information on where/why the name `LDR_EXTERNAL_DLL_PATH` was found/chosen. The "external" seems like an odd choice, the resource could be internal to my own (and even current) module.

Reply:
I'll restore it back to when there was a helper union. As for `LDR_EXTERNAL_DLL_PATH`, see https://github.com/winsiderss/systeminformer/pull/2547#discussion_r2078760564